### ISSUE TITLE
cannot convert from f16 to i32 when targeting loongarch64

### ISSUE THREAD

**Issue by nektro** [ziglang/zig#21336]
### Zig Version

0.13.0

### Steps to Reproduce and Observed Behavior

```zig
const std = @import("std");

test {
    var a: f16 = 42.0;
    _ = &a;

    var b: i32 = @intFromFloat(a);
    _ = &b;

    try std.testing.expect(b == 42);
}
```

```
$ zig test test.zig -target loongarch64-linux-musl
LLVM ERROR: Cannot select: 0x7fb49a509b20: i64 = fp_to_fp16 0x7fb49956ba80
  0x7fb49956ba80: f32,ch = CopyFromReg 0x7fb4999973e0, Register:f32 %11
    0x7fb49956f010: f32 = Register %11
In function: __fixhfsi
Aborted (core dumped)
```

### Expected Behavior

all 1 tests passed

**Comment by nektro** [comment]
can confirm this also affects 0.14.0-dev.1504+5f3d9e0b7

**Comment by alexrp** [comment]
https://github.com/llvm/llvm-project/issues/93894

**Comment by nektro** [comment]
nice, https://github.com/llvm/llvm-project/commit/d075b7bbace8c0ef983ea6f9aad175bec3ede729 made it into llvm 19

**Comment by alexrp** [comment]
Now for the not-so-nice news: https://github.com/ziglang/zig-bootstrap/issues/164#issuecomment-2332357069

**Comment by yxd-ym** [comment]
https://github.com/ziglang/zig-bootstrap/issues/164#issuecomment-2336536410

managed to pass this test with zig bootstrapped for loongarch64-linux-gnu in comment above.

```
yxd@derp ~/tmp/test_zig $ file ~/opt/zig/musl-19/zig
/home/yxd/opt/zig/musl-19/zig: ELF 64-bit LSB executable, LoongArch, version 1 (GNU/Linux), statically linked, stripped
yxd@derp ~/tmp/test_zig $ ~/opt/zig/musl-19/zig version
0.14.0-dev.1499+0006b56f4
yxd@derp ~/tmp/test_zig $ cat test.zig
const std = @import("std");

test {
    var a: f16 = 42.0;
    _ = &a;

    var b: i32 = @intFromFloat(a);
    _ = &b;

    try std.testing.expect(b == 42);
}
yxd@derp ~/tmp/test_zig $ ~/opt/zig/musl-19/zig test test.zig -target loongarch64-linux-musl
All 1 tests passed.
```
```

**Comment by yxd-ym** [comment]
Now try to fix this issue with

https://github.com/llvm/llvm-project/pull/107791

**Comment by andrewrk** [comment]
Fixed with the merge of LLVM 19 branch in c6ad4521c77372cb52adfb9a52b0854d830fed9c.
